generate_workspace/src/test/java/com/google/devtools/build/workspace/output/BzlWriterTest.java

```diff
@@ -35,18 +42,33 @@
   @Test
   public void writeEmpty() throws Exception {
     BzlWriter writer = new BzlWriter(new String[]{}, System.getenv("TEST_TMPDIR"));
-    writer.write(ImmutableList.of());
+    writer.write(createRules());
     String fileContents = Files.toString(
         new File(System.getenv("TEST_TMPDIR") + "/generate_workspace.bzl"),
         Charset.defaultCharset());
     assertThat(fileContents).contains("def generated_maven_jars():\n  pass\n");
     assertThat(fileContents).contains("def generated_java_libraries():\n  pass\n");
   }
 
+  /** Ensures that it automatically creates output directory if they do not exist */
+  @Test
+  public void automaticallyCreateParentDirectories() throws Exception {
+    FileSystem fs = Jimfs.newFileSystem(Configuration.unix());
+    Path root = fs.getPath("root");
+    createDirectories(root);
+
+    Path outputDir = root.resolve("child/directory/");
+
+    BzlWriter writer = new BzlWriter(new String[]{}, outputDir.toString());
+    writer.write(createRules("x:y:1.2.3"));
+
+    assertThatFileExists(outputDir.resolve("generate_workspace.bzl"));
+  }
```

Comment:
Done. The additional test tries outputting to `/root/something/` which fails since Jimfs doesn't allow creating files using absolute paths. 

What exactly are you wanting to test? I simply assert that `/root/something/generate_workspace.bzl` does not exist. However, the nonexistence of this file could be for a host of reasons. For example, if we change BzlWriter to write to `fancy_generate_workspace.bzl` this test would still pass, but it wouldn't communicate anything of value.